gobblin-core/src/main/java/gobblin/instrumented/extractor/InstrumentedExtractorDecorator.java

```diff
@@ -71,4 +71,14 @@ public long getExpectedRecordCount() {
   public long getHighWatermark() {
     return this.embeddedExtractor.getHighWatermark();
   }
+
+  @Override
+  public void close()
```

Comment:
You're right the embedded constructs are not being closed. However, another way of closing them would be to call in the constructor:
this.embeddedExtractor = this.closer.register(extractor);

Then they would be closed automatically by the super.close() method.